yarn-project/prover-node/src/prover-coordination/factory.ts

```diff
@@ -1,10 +1,60 @@
-import { type ProverCoordination, createAztecNodeClient } from '@aztec/circuit-types';
+import { type ArchiveSource, type Archiver } from '@aztec/archiver';
+import { BBCircuitVerifier, TestCircuitVerifier } from '@aztec/bb-prover';
+import {
+  type AztecNode,
+  type ProverCoordination,
+  type WorldStateSynchronizer,
+  createAztecNodeClient,
+} from '@aztec/circuit-types';
+import { createDebugLogger } from '@aztec/foundation/log';
+import { type P2PClient, createP2PClient } from '@aztec/p2p';
+import { type TelemetryClient } from '@aztec/telemetry-client';
 
-import { type ProverCoordinationConfig } from './config.js';
+import { type ProverNodeConfig } from '../config.js';
+
+// We return a reference to the P2P client so that the prover node can stop the service when it shuts down.
+type ProverCoordinationWithP2P = [ProverCoordination, P2PClient | undefined];
```

Comment:
Same thing here- can we just hide from the prover node that it has a P2PClient, and always return a `ProverCoordination`?